doc/src/build/dev_cheat_sheet.md

```diff
@@ -17,7 +17,7 @@ Quick reference on best practices for Sui Network developers.
 - Use `vector`-backed collections (`vector`, `VecSet`, `VecMap`, `PriorityQueue`) with a **known** maximum size of ≤ 1000 items.
     - Use dynamic field-backed collections (`Table`, `Bag`, `ObjectBag`, `ObjectTable`, `LinkedTable`) for any collection that allows third-party addition, larger collections, and collections of unknown size.
     - Sui Move objects have a maximum size of 250KB—any attempt to create a larger object will lead to an aborted transaction. Ensure that your objects do not have an ever-growing `vector`-backed collection.
-- If your function `f` needs a payment in (e.g.) SUI from the caller, use `fun f(payment: Coin<SUI>)` not `fun f(payment: &mut Coin<SUI>, amount: u64)`. This is safer for callers—they know exactly how much they are paying, and do not need to trust `f` to extract the right amount.
+- If your function `f` needs a payment in (e.g.) SUI from the caller, use `fun f(payment: &mut Coin<SUI>, amount: u64)` not `fun f(payment: Coin<SUI>)`. This is safer for callers—they know exactly how much they are paying, and do not need to trust `f` to extract the right amount.
```

Comment:
I'm rusty on what our latest recommendations are here, so will leave @sblackshear and/or @damirka to comment on what the recommended approach is.

My take is abstract because I don't find myself writing this kind of code as often as the folks I've cc'ed, but my preference is for a `fun f(payment: &mut Coin<SUI>)` interface, with the expectation that `f` takes the expected amount from the `payment` coin (so it's a hybrid between the two approaches here).

The reason for this is primarily flexibility -- like with gas, for some applications it's not easy to know up-front what the payment amount is going to be.  This API allows you to split off a coin to act as a budget that `f` can then (internally) split off the actual payment from, leaving the rest in place.

Reply:
I don't have a strong opinion on the topic. Overall I prefer an owned version given that we can check the "price" or some expected parameter before running the call. However, most wallets show how much a transaction would cost, and if anything unexpected happens, user will see it (at least for SUI coin).

Reply:
Also, if this is changed, the doc should be updated accordingly.